### DISCUSSION TITLE
Как выгрузить кастомную csv?

### DISCUSSION THREAD

**Discussion by Vadimgrig2012** [totumonline/totum-ru-issues-and-discussions#298] [Как это сделать?] [answered]
Всем добрый день. Суть вопроса следующая: необходимо сделать кастомную кнопку при нажатии на которую будет выгружаться csvшка с выбранными ранее пользователем строками и с определенными условием полями. Знаю, что есть linkToFileDownload, но она скачивает заранее подготовленный файл . И тут вопрос в том, как сначала создать csvшку в необходимом формате, а уже потом ее выгрузить и лучше конечно, если файл не будет сохранен в БД...

**Comment by erpic-ru** [comment, marked as answer]
Как быстрое решение – сделайте временную таблицу и ее выгружайте через стандартные инструменты

**Reply by Vadimgrig2012** [reply]
Спасибо. Как вариант. Но хотелось бы конечно более "красивое" решение.

**Reply by ttm-alexeypolunin** [reply]
Собственно это норм идея. Немного ее докрутим.

Мы не будем пользователю показывать эту временную таблицу, а заберем из нее данные.

Те в колонки строчной части мы заполняем через `data` в `linkToDataTable`.

В поле `rezultat_po_stroke` мы клеим через `;`:

```
= : listJoin(list: $list; str: ";")

	list: listCreate(item: #pole_1; item: #pole_2)
```

Когда мы передаем `hide: true` то вместо показа пользователю мы получаем хэш по которому берем данные и также клеим этот список используя как соединитель символ переноса строки `$#nl`.

```
= : linkToFileDownload(file: $file)

file: rowCreate(field: "name" = "имя файла.csv"; field: "type" = ".csv"; field: "filestring" = $text)

text: listJoin(list: $list; str: $#nl)

	list: selectList(table: 'vygruzka_csv'; hash: $hash; field: 'rezultat_po_stroke')
		hash: linkToDataTable(table: 'vygruzka_csv'; data: $rowList; hide: true)
			rowList: rowListCreate(field: "pole_1" = json`[1,2,3]`; field: "pole_2" = json`["a","b","c"]`)
```
